src/containers/Layout/ScrollToTop/index.tsx

```diff
@@ -0,0 +1,35 @@
+import { ButtonWrapper, Icon } from 'components';
+import { useEffect, useState } from 'react';
+
+const ScrollToTop = () => {
+  const [isVisible, setIsVisible] = useState(false);
+  const viewportHeight = window.innerHeight;
+  const scrollElem = document.getElementById('page-container');
+
+  useEffect(() => {
+    const toggleVisibility = () => {
+      if (scrollElem?.scrollTop && scrollElem.scrollTop > viewportHeight) {
+        setIsVisible(true);
+      } else {
+        setIsVisible(false);
+      }
+    };
+
+    scrollElem?.addEventListener('scroll', toggleVisibility);
+
+    return () => scrollElem?.removeEventListener('scroll', toggleVisibility);
+  }, []);
+
+  return (
+    isVisible && (
+      <ButtonWrapper
+        className="fixed bottom-16 right-2 h-10 w-10 p-0 rounded-full border-0 bg-lightGrey shadow-scrollToTop lg:hidden"
+        onClick={() => scrollElem?.scrollTo({ behavior: 'smooth', top: 0 })}
+      >
```

Comment:
A small design tweak: you could use css to show/hide the button in order to animate its entrance and exit, while using `pointer-events-none` when it's hidden to prevent it from being clicked.

```suggestion
      <ButtonWrapper
      className={cn(
        'shadow-scrollToTop fixed bottom-16 right-2 h-10 w-10 rounded-full border-0 bg-lightGrey p-0 transition-opacity lg:hidden',
        isVisible ? 'opacity-1' : 'pointer-events-none opacity-0',
      )}
      onClick={() => scrollElem?.scrollTo({ behavior: 'smooth', top: 0 })}
    >
```